### PR TITLE
Seclink tool

### PR DESCRIPTION
**Pull request by pillip8282**

### CODE REVIEW

File: apps/examples/security_test/sltool/Makefile [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+#
+###########################################################################
+
+-include $(TOPDIR)/.config
+-include $(TOPDIR)/Make.defs
+include $(APPDIR)/Make.defs
+
+APPNAME = sltool
+FUNCNAME = $(APPNAME)_main
+THREADEXEC = TASH_EXECMD_ASYNC
```
Comment (jeongchanKim) [inline apps/examples/security_test/sltool/Makefile:25]: Could you add `PRIORITY` and `STACKSIZE` also?

File: apps/examples/security_test/sltool/seclink_tool_main.c [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <pthread.h>
```
Comment (jeongchanKim) [inline apps/examples/security_test/sltool/seclink_tool_main.c:25]: I think this is not needed.

File: apps/examples/security_test/sltool/sltool_key_cmd.c [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+int sltool_handle_remove_key(sl_options_s *opt)
+{
+	printf("[pkbuild] idx %d  %s:%d\n", opt->idx, __FUNCTION__, __LINE__);
+	return 0;
+}
+
+int sltool_handle_gen_key(sl_options_s *opt)
+{
+printf("[pkbuild] idx %d  %s:%d\n", opt->idx, __FUNCTION__, __LINE__);
```
Comment (jeongchanKim) [inline apps/examples/security_test/sltool/sltool_key_cmd.c:30]: Please check the indentation.

File: apps/examples/security_test/sltool/Makefile (hunk 1 above)
Comment (pillip8282) [reply apps/examples/security_test/sltool/Makefile:25]: Ok I'll add those configurations

File: apps/examples/security_test/sltool/seclink_tool_main.c (hunk 2 above)
Comment (pillip8282) [reply apps/examples/security_test/sltool/seclink_tool_main.c:25]: I'll remove pthread.h

File: apps/examples/security_test/sltool/sltool_key_cmd.c (hunk 3 above)
Comment (pillip8282) [reply apps/examples/security_test/sltool/sltool_key_cmd.c:30]: Thanks I'll modify it

File: apps/examples/security_test/sltool/sltool_auth_cmd.c [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+	if ((res = sl_init(&hnd)) != SECLINK_OK) {
+		printf("[sltool] error to sl_init %d\n", res);
+		return -1;
+	}
+	if ((res = sl_remove_certificate(hnd, opt->idx)) != SECLINK_OK) {
+		printf("[sltool] error %s %s:%d\n", sl_strerror(res), __FUNCTION__, __LINE__);
+		fres = -1;
+		goto out;
+	}
+out:
```
Comment (Taejun-Kwon) [inline apps/examples/security_test/sltool/sltool_auth_cmd.c:57]: it seems that unnecessary goto state.

File: apps/examples/security_test/sltool/sltool_auth_cmd.c [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+	input.data = g_test_crt;
+	input.data_len = sizeof(g_test_crt);
+	input.priv = NULL;
+	input.priv_len = 0;
+	if ((res = sl_set_certificate(hnd, opt->idx, &input)) != SECLINK_OK) {
+		printf("[sltool] error %s %s:%d\n", sl_strerror(res), __FUNCTION__, __LINE__);
+		fres = -1;
+		goto out;
+	}
+out:
```
Comment (Taejun-Kwon) [inline apps/examples/security_test/sltool/sltool_auth_cmd.c:86]: same as above.

File: apps/examples/security_test/sltool/sltool_auth_cmd.c (hunk 4 above)
Comment (pillip8282) [reply apps/examples/security_test/sltool/sltool_auth_cmd.c:57]: You're right. Thanks I'll modify it

**Review by an4967 — APPROVED** [review]
LGTM

**Pull request merged**